spec/std/yaml/from_yaml_spec.cr

```diff
@@ -0,0 +1,359 @@
+      end
+    end
+
+    it "should raise when not a number" do
+      expect_raises(YAML::ParseException) do
+        type.from_yaml("hi")
+      end
+    end
+  end
+end
+
+[Float32, Float64].each do |type|
+  describe "#{type.name}.from_yaml" do
+    it "should serialize and deserialize" do
+      type.from_yaml(type.new("1.0").to_yaml).should eq type.new("1.0")
+    end
+
+    it "should parse a number into an {{type.id}}" do
+      type.from_yaml("1").should eq type.new("1")
+    end
+
+    it "should parse a float into an {{type.id}}" do
+      type.from_yaml("1.1").should eq type.new("1.1")
+    end
+
+    it "should raise if quoted with double quotes" do
+      expect_raises(YAML::ParseException) do
+        type.from_yaml %("1.1")
+      end
+    end
+
+    it "should raise if quoted with single quotes" do
+      expect_raises(YAML::ParseException) do
+        type.from_yaml "'1.1'"
+      end
+    end
+
+    it "should raise when not a float or number" do
+      expect_raises(YAML::ParseException) do
+        type.from_yaml("hi")
+      end
+    end
+  end
+end
+
+describe "String.from_yaml" do
+  it "should serialize and deserialize" do
+    String.from_yaml("hi".to_yaml).should eq "hi"
+  end
+
+  it "should parse a string" do
+    String.from_yaml("hello").should eq "hello"
+  end
+
+  it "should parse a single quoted string" do
+    String.from_yaml("'hello'").should eq "hello"
+  end
+
+  it "should parse a double quoted string" do
+    String.from_yaml(%("hello")).should eq "hello"
+  end
+
+  it "should parse a literal string" do
+    String.from_yaml("|\n  hello").should eq "hello"
+  end
+
+  it "should parse a folded string" do
+    String.from_yaml(">\n  hello").should eq "hello"
+  end
+
+  context "reserved values" do
+    values = {
+      "true", "True", "TRUE", "on", "On", "ON", "y", "Y", "yes", "Yes", "YES",
+      "false", "False", "FALSE", "off", "Off", "OFF", "n", "N", "no", "No", "NO",
+      "", "NULL", "Null", "null", "~",
+      ".inf", ".Inf", ".INF", "+.inf", "+.Inf", "+.INF",
+      "-.inf", "-.Inf", "-.INF",
+      ".nan", ".NaN", ".NAN",
+      "+1", "1", "-1",
+      "1.1", "+1.1", "-1.1",
+      "0x_0A_74_AE", "0b1010_0111_0100_1010_1110", "02472256",
+      "2001-12-15T02:59:43.1Z",
+    }
+    values.each do |value|
+      it %(should raise if a reserved value "#{value}") do
+        expect_raises(YAML::ParseException) do
+          String.from_yaml(value)
+        end
+      end
+
+      it "should parse if a reserved value is quoted with double quotes" do
+        String.from_yaml(%("#{value}")).should eq value
+      end
+
+      it "should parse if a reserved value is quoted with single quotes" do
+        String.from_yaml("'#{value}'").should eq value
+      end
+    end
+  end
+end
+
+describe "Array.from_yaml" do
+  it "should serialize and deserialize" do
+    Array(String).from_yaml(["hi"].to_yaml).should eq ["hi"]
+  end
+
+  it "it should an array of the correct objects" do
```

Comment:
- I think the leading `it` in the String is superflous.
- A verb is missing here -> `should parse an array ...` maybe?